src/main/java/org/datadog/jenkins/plugins/datadog/trace/DatadogTraceCache.java

```diff
@@ -0,0 +1,15 @@
+package org.datadog.jenkins.plugins.datadog.trace;
+
+import io.opentracing.Span;
+import org.datadog.jenkins.plugins.datadog.util.SuppressFBWarnings;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public final class DatadogTraceCache {
+    // TODO: Make this a expiring cache with proper settings
+    // TODO: make it thread safe using a ConcurrentMap
+    @SuppressFBWarnings("MS_SHOULD_BE_FINAL")
+    public static Map<String, Span> cache = new HashMap<>();
```

Comment:
Reduce risk of a memory leak by making this a `Map<WeakReference<Run>, Span>` map.

(I was going to suggest using `BuildData`, but it appears that isn't provided to you by the job, but are creating it in the plugin.)

Reply:
I will change this later as mentioned in my TODOs. I don't want a reference on the Run object but instead a string representation of a running build. For now, i just want this POC to work with a very simple scenario.